managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/cache/RangeEntryCacheImpl.java

```diff
@@ -257,6 +265,79 @@ public void asyncReadEntry(ReadHandle lh, long firstEntry, long lastEntry, boole
         }
     }
 
+    @AllArgsConstructor
+    private static final class ReadEntriesCallbackWithContext {
+        ReadEntriesCallback callback;
+        Object ctx;
+    }
+    private class CachedPendingRead {
+        final Triple<Long, Long, Long> key;
```

Comment:
It's better to use a custom Lombok `@Value` class for this purpose instead of Triple<Long, Long, Long>. `Long` is a wrapper for long so there's multiple levels of wrapper objects for the key.

Reply:
Makes sense. I will update the patch